payjoin/tests/integration.rs

```diff
@@ -375,30 +375,36 @@ mod integration {
                     &enrolled.fallback_target(),
                 )?;
 
+                // Poll receive request
+                let (req, ctx) = enrolled.extract_req()?;
+                let response = agent.post(req.url).body(req.body).send().await?;
+                assert!(response.status().is_success());
+                let response_body =
+                    enrolled.process_res(response.bytes().await?.to_vec().as_slice(), ctx).unwrap();
+                // No proposal yet since sender has not responded
+                assert!(response_body.is_none());
+
                 // **********************
                 // Inside the Sender:
                 // Create a funded PSBT (not broadcasted) to address with amount given in the pj_uri
                 let pj_uri = Uri::from_str(&pj_uri_string).unwrap().assume_checked();
                 let psbt = build_original_psbt(&sender, &pj_uri)?;
+                let mut req_ctx = RequestBuilder::from_psbt_and_uri(psbt.clone(), pj_uri.clone())?
+                    .build_with_additional_fee(
+                        Amount::from_sat(10000),
+                        None,
+                        FeeRate::ZERO,
+                        false,
+                    )?;
                 let (Request { url, body, .. }, send_ctx) =
-                    RequestBuilder::from_psbt_and_uri(psbt, pj_uri)?
-                        .build_with_additional_fee(
-                            Amount::from_sat(10000),
-                            None,
-                            FeeRate::ZERO,
-                            false,
-                        )?
-                        .extract_v2(directory.to_owned())?;
-                let response = agent
-                    .post(url.clone())
-                    .header("Content-Type", payjoin::V1_REQ_CONTENT_TYPE)
```

Comment:
This header didn't seem necessary so I removed it

Reply:
Is text/plain set automatically by reqwest? The header is required by BIP 78 but it's probably added automatically.

https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#protocol

Reply:
I missed that requirement. It doesn't look like `Content-Length` is being set anywhere though, despite also being required by BIP78.

Reply:
Added it back - explicit is better

Reply:
Content length is for sure automatic with reqwest. process_response functions test both at least in v1